dataset/dataset_operations_common.h

```diff
@@ -58,76 +67,51 @@ DataArray apply_and_drop_dim_impl(const DataArrayConstView &a, Func func,
 
   if constexpr (ApplyToData) {
     if (a.hasData()) {
-      return DataArray(func(a.data(), dim, std::forward<Args>(args)...),
-                       std::move(coords), std::move(masks), std::move(attrs),
-                       a.name());
+      return DataArray(func(a.data(), dim, args...), std::move(coords),
+                       std::move(masks), std::move(attrs), a.name());
     } else {
-      return DataArray(
-          func(a.dims(), a.unaligned(), dim, std::forward<Args>(args)...),
-          std::move(coords), std::move(masks), std::move(attrs), a.name());
+      if constexpr (std::is_base_of_v<no_realigned_support_t, Func>)
+        throw std::logic_error("Operation cannot handle realigned data.");
```

Comment:
Slight orthogonal - but is it worth now making some notes about this in https://scipp.github.io/user-guide/unaligned-data.html ?

Reply:
Probably, but first we should understand which operations cannot be supported, vs. which operations are just not implemented yet.